tuner/tuner/dispatch_constraints.py

```diff
@@ -27,8 +27,15 @@ def get_mfma_intrinsic_constraints(
     assert len(compatible_intrinsics) > 0, "No compatible intrinsics found"
     return z3.Or(
         *(
-            z3.And(intrinsic_m == mfma.m, intrinsic_n == mfma.n, intrinsic_k == mfma.k)
-            for mfma in compatible_intrinsics
+            z3.And(
+                intrinsic_m == mnk[0],
+                intrinsic_n == mnk[1],
+                intrinsic_k == mnk[2],
+            )
+            for mma_attr in (
+                iree_gpu.MMAAttr.get(mfma) for mfma in compatible_intrinsics
+            )
+            for mnk in [mma_attr.mnk_shape]
```

Comment:
Instead of a for loop, can we make it a local variable? Double for loop with just one iteration is confusing to me -- it took me a good minute to figure out what we are trying to do here.